### ISSUE TITLE
Training Performance is poor

### ISSUE THREAD

**Issue by adeelurehman** [eelyort/shrewdSnek2#2]
Use cwrap or ccall, and write training functions in C or C++ to optimize performance.

**Comment by eelyort** [comment]
Yeah, C++ is a much better idea than JS for this, thanks for the suggestion and info about the libraries. For now, I'd like to keep it fully JS though. I still haven't used web workers for true multithreaded JS too.
